AbstractClient.php

```diff
@@ -124,6 +161,12 @@ protected function buildOptions(Endpoint\EndpointInterface $endpoint) : array
             $options['form_params'] = $formData;
         }
 
+        if (count($this->customHeaders) > 0) {
+            $options['headers'] = $this->customHeaders;
+
+            $this->customHeaders = [];
+        }
```

Comment:
Instead of introducing a new property I would use the optionBuilder here. But we should change it a bit, that it doesn't return if the $options !== null and add extra constraints to the if's above, so we don't override the query/json/form_params options if they are already set.

Reply:
Yes, that is the reason why I added a new property. I will refactor it a bit.